spyder/widgets/ipythonconsole/client.py

```diff
@@ -378,7 +387,20 @@ def restart_kernel(self):
     @Slot(str)
     def kernel_restarted_message(self, msg):
         """Show kernel restarted/died messages."""
-        stderr = codecs.open(self.stderr_file, 'r', encoding='utf-8').read()
+        try:
+            stderr = codecs.open(self.stderr_file, 'r',
+                                 encoding='utf-8').read()
+        except UnicodeDecodeError:
+            # This handling is needed since the stderr file could be encoded
+            # in something different to utf-8. In case of fail at least
+            # we raise an informative exception.
+            # See issue 4191
+            try:
+                stderr_text = open(self.stderr_file, 'rb').read()
+                encoding = get_coding(stderr_text)
+                stderr = to_text_string(stderr_text, encoding)
+            except UnicodeDecodeError:

```

Comment:
Please change this to only `except` here. We want to catch all errors at this point.